runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkExecutionEnvironments.java

```diff
@@ -77,17 +76,23 @@ static ExecutionEnvironment createBatchExecutionEnvironment(
       flinkBatchEnv = new CollectionEnvironment();
     } else if ("[auto]".equals(masterUrl)) {
       flinkBatchEnv = ExecutionEnvironment.getExecutionEnvironment();
-    } else if (masterUrl.matches(".*:\\d*")) {
-      List<String> parts = Splitter.on(':').splitToList(masterUrl);
+    } else {
+      String[] hostAndPort = masterUrl.split(":", 2);
+      final String host = hostAndPort[0];
+      final int port;
+      if (hostAndPort.length > 1) {
+        port = Integer.parseInt(hostAndPort[1]);
```

Comment:
Shall we catch the integer parsing exception and log 
`     LOG.warn("Unrecognized Flink Master URL {}", masterUrl);	`

Reply:
If the port can't be parsed, then something is wrong. I would just throw an exception with a meaningful error message. 

As a side note, I realized we don't support IPv6 Ip addresses.